pkgs/applications/editors/vscode/extensions/vscode-lldb/default.nix

```diff
@@ -52,6 +52,14 @@ let
     dontNpmInstall = true;
   }));
 
+  # debugservers on macOS require the 'com.apple.security.cs.debugger'
+  # entitlement which nixpkgs' lldb-server does not yet provide; see
+  # <https://github.com/NixOS/nixpkgs/pull/38624> for details
+  lldbServer = if stdenv.isDarwin then
+    "/Applications/Xcode.app/Contents/SharedFrameworks/LLDB.framework/Versions/A/Resources/debugserver"
```

Comment:
That would require XCode to be installed, sadly...

Reply:
Yep — absent a way to get the necessary entitlement on a debugserver provided by nixpkgs, Xcode (or another suitably signed binary) is required. 

That said, maybe there’s a way to prompt users to install it if needed that I’ve missed — do you see one?

(and in any event, thanks for looking this over + for sharing feedback!)